### PR TITLE
Add policy.open-cluster-management.io healthchecks

### PR DESCRIPTION
**Pull request by JustinKuli**
This is a pre-community PR for squad review.

### CODE REVIEW

File: resource_customizations/policy.open-cluster-management.io/ConfigurationPolicy/health_test.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      message: >-
+        notification - namespaces [argo-example] was created successfully;
+        notification - namespaces [argo-example-2] was created successfully
+    inputPath: testdata/healthy_created.yaml
+  - healthStatus:
+      status: Healthy
+      message: >-
+        notification - namespaces [argo-example] found as specified;
+        notification - namespaces [argo-example-2] found as specified
+    inputPath: testdata/healthy_found.yaml
```
Comment (yiraeChristineKim) [inline resource_customizations/policy.open-cluster-management.io/ConfigurationPolicy/health_test.yaml:27]: Optional: Can we add for "Waiting for compliance" and "Progressing"

File: resource_customizations/policy.open-cluster-management.io/OperatorPolicy/health.lua [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      hs.status = "Healthy"
+      return hs
+    else
+      hs.status = "Degraded"
+      return hs
+    end
+  end
+end
+hs.status = "Progressing"
+hs.message = "Waiting for the compliance condition"
```
Comment (yiraeChristineKim) [inline resource_customizations/policy.open-cluster-management.io/OperatorPolicy/health.lua:25]: Optional: Do we need a test for "Waiting for the compliance condition"

**Review by yiraeChristineKim — APPROVED** [review]
Great work!

File: resource_customizations/policy.open-cluster-management.io/ConfigurationPolicy/health_test.yaml (hunk 1 above)
Comment (JustinKuli) [reply resource_customizations/policy.open-cluster-management.io/ConfigurationPolicy/health_test.yaml:27]: There is one for Progressing, I don't think "Waiting for compliance" is very likely, the policy would need to set status.compliant but not status.compliancyDetails

File: resource_customizations/policy.open-cluster-management.io/OperatorPolicy/health.lua (hunk 2 above)
Comment (JustinKuli) [reply resource_customizations/policy.open-cluster-management.io/OperatorPolicy/health.lua:25]: I'll add that one, good idea

**Pull request closed without merging**